### PR TITLE
Add RootDetAtom

### LINKED ISSUE #214: Support for det(P)^(1/n)
In trying to solve a problem with objective function
`maximize det(P)`
there are two common transformations that turn the objective into a convex objective.
The first is to convert the problem to
`maximize logdet(P)`
which is currently implemented and can be solved using e.g. SCS.
Another option is to instead solve
`maximize det(P)^(1/n)`
This is equivalent to the geometric mean of the eigenvalues of `P`. From what I've read, this can be modeled with semidefinite and second order cones (I don't know the details of this). This allows a lot more solvers to be used to solve this optimization problem (instead of relying on a first order method like SCS)

Would it be possible to get support for the nth root of the determinant added?

### PR DESCRIPTION
**Pull request by odow**
Closes #214

### CODE REVIEW

File: docs/src/manual/operations.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ solver (including SCS and Mosek).
 | `opnorm(x, 2)` (`operatornorm(x)`)| max of singular values of $x$      | convex       | not monotonic |                        |
 | `eigmax(x)`                       | max eigenvalue of $x$              | convex       | not monotonic |                        |
 | `eigmin(x)`                       | min eigenvalue of $x$              | concave      | not monotonic |                        |
+| `rootdet(X)`                      | root-determinant of the matrix X   | concave      | not monotonic |                        |
```
Comment (blegat) [inline docs/src/manual/operations.md:116]: Maybe mention somewhere that by root we mean the `n`-th root ?

Comment (odow) [reply docs/src/manual/operations.md:116]: CVX calls this `det_rootn`. Is that better?

Comment (blegat) [reply docs/src/manual/operations.md:116]: `rootdet` is more consistent with `LinearAlgebra.logdet`. We could also just support `det(Q)^(1/n)`, that will be needed anyway for the MOI wrapper to support `JuMP.@objective(model, Max, det(Q)^(1/n))`

Comment (blegat) [reply docs/src/manual/operations.md]: We decided to go for `rootdet` in the JuMP-dev call, we can add `rootdet` as operator to JuMP later if needed

**Pull request merged**